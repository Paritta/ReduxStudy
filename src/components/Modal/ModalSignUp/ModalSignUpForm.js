import React from "react";
import styled from "styled-components";
import { Field, reduxForm } from "redux-form";
import oc from "open-color";
import PropTypes from "prop-types";

const propTypes = {
    handleSubmit: PropTypes.func,
    hideModal: PropTypes.func,
};

const defaultTypes = {
    handleSubmit() {},
    hideModal() {},
};

const Wrapper = styled.div`
    padding: 30px 0 10px 0;
`;

const StyledField = styled(Field)`
    background-color: ${oc.gray[0]};
    width: 400px;
    height: 40px;
    font-size: 15px;
    border: 2px solid white;
    padding-left: 20px;
    margin-bottom: 10px;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

export let ModalSignUpForm = ({ handleSubmit, hideModal }) => {
    return (
        <Wrapper>
            <form onSubmit={ handleSubmit }>
                <div>
                    <StyledField name="email" component="input" type="text" required placeholder="email"/>
                </div>
                <div>
                    <StyledField name="Nickname" component="input" type="text" required placeholder="Nickname"/>
                </div>
                <div>
                    <StyledField name="password" component="input" type="password" required placeholder="password"/>
                </div>
                <Button type="submit">회원 가입</Button>
                <Button onClick={() => hideModal()}>나가기</Button>
            </form>
        </Wrapper>
    )
};

ModalSignUpForm.propTypes = propTypes;
ModalSignUpForm.defaultTypes = defaultTypes;

ModalSignUpForm = reduxForm({
    form: 'ModalSignUp'
})(ModalSignUpForm);

export default ModalSignUpForm;