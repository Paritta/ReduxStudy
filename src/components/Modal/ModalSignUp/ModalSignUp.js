import React from "react";
import styled, {keyframes} from "styled-components";
import { bounceInLeft, bounceOutRight } from 'react-animations';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import ModalSignUpForm from "./ModalSignUpForm";
import { firebaseConnect, pathToJS } from "react-redux-firebase";

const propTypes = {
    Modal: PropTypes.object,
    showModal: PropTypes.func,
    hideModal: PropTypes.func,
    Animate: PropTypes.object,
    AnimateTurn: PropTypes.func,
    AnimateDown: PropTypes.func,
    firebase: PropTypes.object,
    fetchRequest: PropTypes.func,
    toppopuReceiveRequest: PropTypes.func,
};

const defaultTypes = {
    Modal: {},
    showModal() {},
    hideModal() {},
    Animate: {},
    AnimateTurn() {},
    AnimateDown() {},
    fetchRequest() {},
    toppopuReceiveRequest() {},
};

const bounceInLeftAnimation = keyframes`${bounceInLeft}`;
const bounceOutRightAnimation = keyframes`${bounceOutRight}`;

const AnimationWrapper = styled.div`
    animation: 0.5s ${bounceInLeftAnimation};
`;

const AnimationOutWrapper = styled.div`
    animation: 0.5s ${props => props.AnimateOut ? bounceOutRightAnimation : ""};
`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

export class ModalSignUp extends React.Component {
    createNewUser = ({ email, password, username }) => {
        this.props.firebase.createUser(
            { email, password },
            { username, email }
        )
        .then(() => {
            this.props.hideModal();
            this.props.fetchRequest();
            this.props.toppopuReceiveRequest();
        })
        .catch(error => {
            console.log(error.code);
            this.props.showModal({ modalType: "MODAL_ERROR" })
        })
};

    submit = values => {
        this.createNewUser({
            email: values.email,
            password: values.password,
            username: values.Nickname,
        })
    };

    render () {
        const { hideModal, animateTurn, animateDown, Animate } = this.props;

        return (
            <div>
                <Dimmed />
                <AnimationOutWrapper
                    AnimateOut={Animate.AnimateOut}
                >
                    <AnimationWrapper>
                        <Wrapper>
                            <ModalHeader>회원가입</ModalHeader>
                            <ModalBox>
                                <ModalSignUpForm
                                    onSubmit={this.submit}
                                    googleLogin={this.googleLogin}
                                    hideModal={hideModal}
                                    animateTurn={animateTurn}
                                    animateDown={animateDown}
                                />
                            </ModalBox>
                        </Wrapper>
                    </AnimationWrapper>
                </AnimationOutWrapper>
            </div>
        )
    }
}

ModalSignUp.propTypes = propTypes;
ModalSignUp.defaultTypes = defaultTypes;

const WrappedModalSignUp = firebaseConnect()(ModalSignUp);

export default connect(
    ({ firebase }) => ({
        authError: pathToJS(firebase, "authError"),
        auth: pathToJS(firebase, "auth"),
        profile: pathToJS(firebase, "profile")
    })
)(WrappedModalSignUp)
