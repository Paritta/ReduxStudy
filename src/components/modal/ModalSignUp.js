import React from "react";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import ModalSignUpForm from "./ModalSignUpForm";
import { firebaseConnect, pathToJS } from "react-redux-firebase";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
    firebase: PropTypes.shape({
        login: PropTypes.func.isRequired
    })
};

const defaultTypes = {
    hideModal() {},
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
`;

const AnimationWrapper = styled.div`
    animation: 1s ${fadeInAnimation};
`;

const ModalHeader = styled.div`
    @import url(//fonts.googleapis.com/earlyaccess/hanna.css);
    font-family: 'Hanna', fantasy;
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

class ModalSignUp extends React.Component {
    createNewUser = ({ email, password, username }) => {
        this.props.firebase.createUser(
            { email, password },
            { username, email }
        )
    };

    submit = values => {
        this.createNewUser({
            email: values.email,
            password: values.password,
            username: values.Nickname
        })
    };

    render () {
        const { hideModal } = this.props;

        return (
            <div>
                <Dimmed />
                    <Wrapper>
                        <AnimationWrapper>
                        <ModalHeader>회원가입</ModalHeader>
                        <ModalBox>
                            <ModalSignUpForm
                                onSubmit={this.submit}
                                hideModal={hideModal}
                            />
                        </ModalBox>
                        </AnimationWrapper>
                    </Wrapper>
            </div>
        )
    }
};

ModalSignUp.propTypes = propTypes;
ModalSignUp.defaultTypes = defaultTypes;

const WrappedModalSignUp = firebaseConnect()(ModalSignUp);

export default connect(
    ({ firebase }) => ({
        authError: pathToJS(firebase, "authError"),
        auth: pathToJS(firebase, "auth"),
        profile: pathToJS(firebase, "profile")
    })
)(WrappedModalSignUp)
