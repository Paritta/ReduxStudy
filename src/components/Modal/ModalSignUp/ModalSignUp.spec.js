import React from "react";
import ReactTestRenderer from "react-test-renderer"
import { ModalSignUp } from "./ModalSignUp";
import "jest-styled-components";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

it("ModalSignUp render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Provider store={store}>
            <ModalSignUp />
        </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
