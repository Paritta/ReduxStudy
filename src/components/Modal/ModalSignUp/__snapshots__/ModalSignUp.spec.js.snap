// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignUp render correctly 1`] = `
.c0 {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 5;
  background: rgba(0,0,0,0.3);
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 10px;
  overflow: hidden;
  z-index: 10;
  width: 550px;
}

.c2 {
  -webkit-animation: 1s eMLfYp;
  animation: 1s eMLfYp;
}

.c3 {
  font-style: normal;
  font-weight: 400;
  color: white;
  padding: 15px 15px;
  background: palevioletred;
}

.c4 {
  background: white;
  border: 1px solid rgba(0,0,0,0.3);
  padding: 15px;
}

.c5 {
  padding: 30px 0 10px 0;
}

.c6 {
  background-color: #f8f9fa;
  width: 400px;
  height: 40px;
  font-size: 15px;
  border: 2px solid white;
  padding-left: 20px;
  margin-bottom: 10px;
}

.c7 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c7:hover {
  background: #f783ac;
  border: 2px solid #f783ac;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        회원가입
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <input
                className="c6"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="email"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div>
              <input
                className="c6"
                name="Nickname"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Nickname"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div>
              <input
                className="c6"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <button
              className="c7"
              type="submit"
            >
              회원 가입
            </button>
            <button
              className="c7"
              onClick={[Function]}
            >
              나가기
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
